### PR TITLE
feat: update AggregateToken

### PR DESCRIPTION
**Pull request by ghost**
## What's new in this PR?

To enable seamless integration between Credbull `LiquidStone` and `AggregateToken`, I made two modifications to `AggregateToken`:
- Since LiquidStone is a MultiTokenVault, AggregateToken must implement `ERC1155Holder` to receive LiquidStone tokens.
- AggregateToken needs a function that calls `requestRedeem` on LiquidStone.

## Why?

What problem does this solve?
Why is this important?
What's the context?

### CODE REVIEW

File: nest/src/AggregateToken.sol [hunk 1]
```diff
@@ -13,7 +14,7 @@ import { IComponentToken } from "./interfaces/IComponentToken.sol";
  * @author Eugene Y. Q. Shen
  * @notice Implementation of the abstract ComponentToken that represents a basket of ComponentTokens
  */
-contract AggregateToken is ComponentToken, IAggregateToken {
+contract AggregateToken is ComponentToken, IAggregateToken, IERC1155Receiver {
```
Comment (eyqs) [inline nest/src/AggregateToken.sol]: Can you use `ERC1155Holder` instead? https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/token/ERC1155/utils/ERC1155Holder.sol

File: nest/src/AggregateToken.sol [hunk 2]
```diff
@@ -199,6 +200,9 @@ contract AggregateToken is ComponentToken, IAggregateToken {
             emit ComponentTokenListed(componentToken);
         }
 
+        // Should approve the componentToken to transfer assets from AggregateToken to ComponentToken.
+        IERC20(asset()).approve(address(componentToken), assets);
```
Comment (eyqs) [inline nest/src/AggregateToken.sol]: Call `approveComponentToken` externally instead

File: nest/src/AggregateToken.sol (hunk 1 above)
Comment (ghost) [reply nest/src/AggregateToken.sol]: Thanks @eyqs 
I fixed

File: nest/src/AggregateToken.sol (hunk 2 above)
Comment (ghost) [reply nest/src/AggregateToken.sol]: @eyqs  I see. We can use `approveComponentToken` externally.
I removed approve.
Thanks

**Review by eyqs — APPROVED** [review]
LGTM

**Pull request merged**